### PR TITLE
chore(js,react): inbox props propagation fix

### PR DESCRIPTION
**Pull request by LetItRock**
### What changed? Why was the change needed?
Fixed the issue with propagating the props changes to the Solid Inbox.

### Screenshots

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

File: packages/js/src/ui/context/InboxContext.tsx [hunk 1]
```diff
@@ -47,6 +47,13 @@ export const InboxProvider = (props: InboxProviderProps) => {
     setFilter((old) => ({ ...old, tags }));
   };
 
+  createEffect(() => {
+    setTabs(props.tabs);
+    const firstTab = props.tabs[0];
+    setActiveTab(firstTab?.label ?? '');
+    setFilter((old) => ({ ...old, tags: firstTab?.value ?? [] }));
+  });
```
Comment (LetItRock) [inline packages/js/src/ui/context/InboxContext.tsx:55]: when the `tabs` prop is changing we need to update the filter and active tab state

File: packages/react/src/components/Inbox.tsx [hunk 2]
```diff
@@ -27,7 +33,7 @@ const DefaultInbox = (props: DefaultInboxProps) => {
         element,
       });
     },
-    [renderNotification, renderBell, onNotificationClick, onPrimaryActionClick, onSecondaryActionClick]
+    [open, renderNotification, renderBell, onNotificationClick, onPrimaryActionClick, onSecondaryActionClick]
```
Comment (LetItRock) [inline packages/react/src/components/Inbox.tsx:36]: whenever these props are changing we are updating the mounted component props

File: packages/react/src/components/Renderer.tsx [hunk 3]
```diff
@@ -38,9 +39,19 @@ export const Renderer = (props: RendererProps) => {
     [setMountedElements]
   );
 
-  React.useEffect(() => {
-    const ui: NovuUI = new NovuUI(options);
-    setNovuUI(ui);
+  useEffect(() => {
+    setNovuUI(new NovuUI(optionsRef.current));
+  }, []);
```
Comment (LetItRock) [inline packages/react/src/components/Renderer.tsx:44]: we should create the NovuUI instance once per component lifecycle

File: packages/react/src/components/Renderer.tsx [hunk 4]
```diff
@@ -38,9 +39,19 @@ export const Renderer = (props: RendererProps) => {
...
+
+  useEffect(() => {
+    if (!novuUI) {
+      return;
+    }
+
+    novuUI.updateAppearance(options.appearance);
+    novuUI.updateLocalization(options.localization);
+    novuUI.updateTabs(options.tabs);
+    novuUI.updateOptions(options.options);
```
Comment (LetItRock) [inline packages/react/src/components/Renderer.tsx:54]: when the props change we need to update the Solid Renderer signals

File: packages/js/src/ui/novuUI.tsx [hunk 5]
```diff
@@ -128,19 +128,19 @@ export class NovuUI {
     });
   }
 
-  updateAppearance(appearance: Appearance) {
+  updateAppearance(appearance?: Appearance) {
```
Comment (SokratisVidros) [inline packages/js/src/ui/novuUI.tsx:131]: For better DX, this shouldn't be optional. Without the appearance object these functions are noop.

File: packages/js/src/ui/novuUI.tsx [hunk 6]
```diff
@@ -128,19 +128,19 @@ export class NovuUI {
...
     this.#setAppearance(appearance);
   }
 
-  updateLocalization(localization: Localization) {
+  updateLocalization(localization?: Localization) {
```
Comment (SokratisVidros) [inline packages/js/src/ui/novuUI.tsx:135]: Ditto.

File: packages/js/src/ui/novuUI.tsx [hunk 7]
```diff
@@ -128,19 +128,19 @@ export class NovuUI {
...
     this.#setLocalization(localization);
   }
 
   updateOptions(options: NovuOptions) {
     this.#setOptions(options);
   }
 
-  updateTabs(tabs: Array<Tab>) {
-    this.#setTabs(tabs);
+  updateTabs(tabs?: Array<Tab>) {
```
Comment (SokratisVidros) [inline packages/js/src/ui/novuUI.tsx:143]: Ditto.

File: packages/react/src/components/Renderer.tsx (hunk 3 above)
Comment (SokratisVidros) [reply packages/react/src/components/Renderer.tsx:44]: Lifecycle as in mount/unmount?

**Review by SokratisVidros — APPROVED** [review]

File: packages/js/src/ui/novuUI.tsx (hunk 5 above)
Comment (LetItRock) [reply packages/js/src/ui/novuUI.tsx:131]: under the hood, we have checks for undefined and fallback to defaults

File: packages/js/src/ui/novuUI.tsx (hunk 6 above)
Comment (LetItRock) [reply packages/js/src/ui/novuUI.tsx:135]: when it's undefined under the hood we set it to the defaults

File: packages/js/src/ui/novuUI.tsx (hunk 7 above)
Comment (LetItRock) [reply packages/js/src/ui/novuUI.tsx:143]: same

File: packages/react/src/components/Renderer.tsx (hunk 3 above)
Comment (LetItRock) [reply packages/react/src/components/Renderer.tsx:44]: yes

**Pull request merged**